orchestrator/test_runner/src/happy_path.rs

```diff
@@ -354,19 +376,26 @@ async fn test_batch(
             .expect("Failed to get current eth valset");
     let starting_batch_nonce = current_eth_batch_nonce;
 
-    let start = Instant::now();
-    while starting_batch_nonce == current_eth_batch_nonce {
-        info!(
-            "Batch is not yet submitted {}>, waiting",
-            starting_batch_nonce
-        );
-        current_eth_batch_nonce =
-            get_tx_batch_nonce(gravity_address, erc20_contract, *MINER_ADDRESS, &web30)
-                .await
-                .expect("Failed to get current eth tx batch nonce");
-        delay_for(Duration::from_secs(4)).await;
-        if Instant::now() - start > TOTAL_TIMEOUT {
-            panic!("Failed to submit transaction batch set");
+    match tokio::time::timeout(TOTAL_TIMEOUT, async {
+        while starting_batch_nonce == current_eth_batch_nonce {
+            info!(
+                "Batch is not yet submitted {}>, waiting",
+                starting_batch_nonce
+            );
+            current_eth_batch_nonce =
+                get_tx_batch_nonce(gravity_address, erc20_contract, *MINER_ADDRESS, &web30)
+                    .await
+                    .expect("Failed to get current eth tx batch nonce");
+            tokio::time::sleep(std::time::Duration::from_secs(4)).await;
+        }
+    })
+    .await
+    {
+        Ok(_) => {
+            println!("Submitted trtransaction batch set");
```

Comment:
Nit: typo on "transaction" here and in the `Err` match arm.

Reply:
These commits fixed happy_path.rs
https://github.com/PeggyJV/gravity-bridge/pull/274/commits/d2dd8d28c91c23058ceef50b56d326841c1f1a12

Reply:
https://github.com/PeggyJV/gravity-bridge/pull/274/commits/b79a0833d835eea249ac98f16b8e9250996edb07